packages/licenses/src/getPkgInfo.ts

```diff
@@ -0,0 +1,267 @@
+  getFilePathInCafs,
+  PackageFileInfo,
+  PackageFilesIndex,
+} from '@pnpm/cafs'
+import loadJsonFile from 'load-json-file'
+
+const limitPkgReads = pLimit(4)
+
+export async function readPkg (pkgPath: string) {
+  return limitPkgReads(async () => readPackageJson(pkgPath))
+}
+
+/**
+ * @const
+ * List of typical names for license files
+ */
+const LICENSE_FILES = ['LICENSE', 'LICENCE', 'LICENSE.md', 'LICENCE.md']
+
+export interface LicenseInfo {
+  name: string
+  licenseFile?: string
+}
+
+/**
+ * Coerce the given value to a string or a null value
+ * @param field the string to be converted
+ * @returns string | null
+ */
+function coerceToString (field: unknown): string | null {
+  const string = String(field)
+  return typeof field === 'string' || field === string ? string : null
+}
+
+/**
+ * Parses the value of the license-property of a
+ * package manifest and return it as a string
+ * @param field the value to parse
+ * @returns string
+ */
+function parseLicenseManifestField (field: unknown) {
+  if (Array.isArray(field)) {
+    const licenses = field
+    const licenseTypes = licenses.reduce((listOfLicenseTypes, license) => {
+      const type = coerceToString(license.type)
+      if (type) {
+        listOfLicenseTypes.push(type)
+      }
+      return listOfLicenseTypes
+    }, [])
+
+    if (licenseTypes.length > 1) {
+      const combinedLicenseTypes = licenseTypes.join(' OR ') as string
+      return `(${combinedLicenseTypes})`
+    }
+
+    return licenseTypes[0] ?? null
+  } else {
+    return (field as { type: string })?.type ?? coerceToString(field)
+  }
+}
+
+/**
+ * Reads the license field or LICENSE file from
+ * the directory of the given package manifest
+ *
+ * If the package.json file is missing the `license`-property
+ * the root of the manifest directory will be scanned for
+ * files named listed in the array LICENSE_FILES and the
+ * contents will be returned.
+ *
+ * @param {*} packageInfo the package to check
+ * @returns Promise<LicenseInfo>
+ */
+async function parseLicense (
+  packageInfo: {
+    manifest: PackageManifest
+    files: Record<string, PackageFileInfo>
+  },
+  opts: { cafsDir: string }
+): Promise<LicenseInfo> {
+  const license = parseLicenseManifestField(packageInfo.manifest.license)
+
+  // check if we discovered a license, if not attempt to parse the LICENSE file
+  if (!license || /see license/i.test(license)) {
+    for (const filename of LICENSE_FILES) {
+      try {
+        const licensePackageFileInfo = packageInfo.files[filename]
+        if (!licensePackageFileInfo) continue
+
+        const licenseContents = await fetchFileFromStore(
+          { integrity: licensePackageFileInfo.integrity },
+          opts
+        )
+
+        return {
+          name: 'Unknown',
+          licenseFile: licenseContents.toString('utf-8'),
+        }
+      } catch (err) {
+        // Silently ignore the error when failed
+        // to read the contents of a license file
```

Comment:
Yeah, that should be fine as it will try to read one of the other license files of this list. 

If none of the files are found it will return `{ name: license ?? 'Unknown' }` on line 156